shark-graph/src/main/java/shark/internal/hppc/Tuples.kt

```diff
@@ -0,0 +1,69 @@
+package shark.internal.hppc
+
+/**
+ * Replacement to a traditional Pair<Long, Object> that doesn't box long.
+ * Class is purposefully not made as a `data class` to decrease memory footprint of the object.
```

Comment:
> not made as a `data class`

=> "not a data class" ?

Reply:
> to decrease memory footprint of the object

Can you outline why a data class would have a larger memory footprint in the javadoc comment?